### PR TITLE
fix(predictions): add adapters for swift sdk clients

### PR DESCRIPTION
**Pull request by atierian**
## Description

- Adds adapters + behavior protocols for AWS SDK for Swift clients to `AWSPredictionsPlugin`.

## General Checklist

- [ ] ~Added new tests to cover change, if needed~
- [ ] ~Build succeeds with all target using Swift Package Manager~
- [ ] ~All unit tests pass~
- [ ] ~All integration tests pass~
- [x] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] ~Documentation update for the change if required~
- [x] PR title conforms to conventional commit style
- [ ] ~New or updated tests include `Given When Then` inline code documentation and are named accordingly `testThing_condition_expectation()`~
- [ ] ~If breaking change, documentation/changelog update with migration instructions~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Predictions/AWSPredictionsPlugin/Dependency/AWSComprehendAdapter.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import AWSComprehend
+
+class AWSComprehendAdapter: AWSComprehendBehavior {
+    let awsComprehend: ComprehendClient
```
Comment (ruisebas) [inline AmplifyPlugins/Predictions/AWSPredictionsPlugin/Dependency/AWSComprehendAdapter.swift:12]: Maybe use `ComprehendClientProtocol` instead? Or is the actual implementation needed for some reason?

File: AmplifyPlugins/Predictions/AWSPredictionsPlugin/Dependency/AWSPollyAdapter.swift [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import AWSPolly
+
+class AWSPollyAdapter: AWSPollyBehavior {
+    let awsPolly: PollyClient
```
Comment (ruisebas) [inline AmplifyPlugins/Predictions/AWSPredictionsPlugin/Dependency/AWSPollyAdapter.swift:12]: `PollyClientProtocol`?

File: AmplifyPlugins/Predictions/AWSPredictionsPlugin/Dependency/AWSRekognitionAdapter.swift [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import AWSRekognition
+
+class AWSRekognitionAdapter: AWSRekognitionBehavior {
+    let awsRekognition: RekognitionClient
```
Comment (ruisebas) [inline AmplifyPlugins/Predictions/AWSPredictionsPlugin/Dependency/AWSRekognitionAdapter.swift:12]: You get the idea 😆

**Review by ruisebas — CHANGES_REQUESTED** [review]

File: AmplifyPlugins/Predictions/AWSPredictionsPlugin/Dependency/AWSComprehendAdapter.swift (hunk 1 above)
Comment (atierian) [reply AmplifyPlugins/Predictions/AWSPredictionsPlugin/Dependency/AWSComprehendAdapter.swift]: TIL about the ClientProtocols. Thanks for pointing that out; that allows us to remove a lot of unnecessary code 🦾 
Addressed in [0e8078c](https://github.com/aws-amplify/amplify-swift/pull/2895/commits/0e8078c01c74b6070d7249a96baecf69690e19d0)

File: AmplifyPlugins/Predictions/AWSPredictionsPlugin/Dependency/AWSRekognitionAdapter.swift (hunk 3 above)
Comment (atierian) [reply AmplifyPlugins/Predictions/AWSPredictionsPlugin/Dependency/AWSRekognitionAdapter.swift]: Addressed in [0e8078c](https://github.com/aws-amplify/amplify-swift/pull/2895/commits/0e8078c01c74b6070d7249a96baecf69690e19d0)

File: AmplifyPlugins/Predictions/AWSPredictionsPlugin/Configuration/ConvertConfiguration.swift [hunk 4]
```diff
@@ -0,0 +1,151 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+//import AWSCore
```
Comment (ruisebas) [inline AmplifyPlugins/Predictions/AWSPredictionsPlugin/Configuration/ConvertConfiguration.swift:8]: Leftover commented out import 😅 

It's also in  `IdentifyConfiguration.swift`, `InterpretConfiguration.swift`, and `PredictionsPluginConfiguration.swift`

**Review by ruisebas — CHANGES_REQUESTED** [review]

File: AmplifyPlugins/Predictions/AWSPredictionsPlugin/Configuration/ConvertConfiguration.swift (hunk 4 above)
Comment (atierian) [reply AmplifyPlugins/Predictions/AWSPredictionsPlugin/Configuration/ConvertConfiguration.swift]: 🤦  _grumble linter grumble grumble_
[07e0cf7](https://github.com/aws-amplify/amplify-swift/pull/2895/commits/07e0cf7cec7c31e2c17e9f38e12c26b6ce935dc3)

**Review by ruisebas — APPROVED** [review]

**Pull request merged**